src/game_state/GameObject.h

```diff
@@ -4,44 +4,43 @@
 #define GAME_OBJECT_H_
 
 #include <memory>
+#include <glm/ext.hpp>
 
 #include "Shape.h"
 #include "Texture.h"
 
 // GameObjects are 3D entities which can be rendered
 class GameObject {
-public:
-   GameObject(std::shared_ptr<Shape> model);
-   virtual ~GameObject();
-
-   std::shared_ptr<Shape> GetModel() {
-      return model;
-   }
-   std::shared_ptr<Texture> GetTexture() {
-      return texture;
-   }
-
-   virtual glm::vec3 GetPosition() {
-      return position;
-   }
-   virtual glm::vec3 GetDirection() {
-      return direction;
-   }
-   float GetScale() {
-      return scale;
-   }
-
-   void SetPosition(glm::vec3 position);
-   void SetDirection(glm::vec3 direction);
-   void SetScale(float scale);
-
-private:
-   std::shared_ptr<Shape> model;
-   std::shared_ptr<Texture> texture;
-
-   glm::vec3 position;
-   glm::vec3 direction;
-   float scale;
+ public:
+  GameObject();
+  GameObject(std::shared_ptr<Shape> model);
+  GameObject(std::shared_ptr<Shape> model, glm::vec3 position);
+  GameObject(std::shared_ptr<Shape> model,
+             glm::vec3 position,
+             glm::vec3 direction,
+             glm::vec3 scale);
```

Comment:
I made only the default no args constructor available since some weird shit happens if I pass a shared pointer to a constructor. Gameobjects should set their model, size, ect, in their own constructors